lib/Compiler.php

```diff
@@ -0,0 +1,195 @@
+use Symfony\Component\Process\Process;
+
+/**
+ * The Compiler class compiles xsd2php into a phar
+ */
+class Compiler
+{
+    private $version;
+    private $versionDate;
+
+    /**
+     * Compiles xsd2php into a single phar file
+     *
+     * @throws \RuntimeException
+     * @param string $pharFile The full path to the file to create
+     */
+    public function compile($pharFile = 'xsd2php.phar')
+    {
+        if (file_exists($pharFile)) {
+            unlink($pharFile);
+        }
+
+        $process = new Process('git log --pretty="%H" -n1 HEAD', __DIR__);
+        if ($process->run() != 0) {
+            throw new \RuntimeException('Can\'t run git log. You must ensure to run compile from xsd2php git repository clone and that git binary is available.');
+        }
+        $this->version = trim($process->getOutput());
+
+        $process = new Process('git log -n1 --pretty=%ci HEAD', __DIR__);
+        if ($process->run() != 0) {
+            throw new \RuntimeException('Can\'t run git log. You must ensure to run compile from xsd2php git repository clone and that git binary is available.');
+        }
+
+        $date = new \DateTime(trim($process->getOutput()));
+        $date->setTimezone(new \DateTimeZone('UTC'));
+        $this->versionDate = $date->format('Y-m-d H:i:s');
+
+        $process = new Process('git describe --tags --exact-match HEAD');
+        if ($process->run() == 0) {
+            $this->version = trim($process->getOutput());
+        }
+
+        $phar = new \Phar($pharFile, 0, 'xsd2php.phar');
+        $phar->setSignatureAlgorithm(\Phar::SHA1);
+
+        $phar->startBuffering();
+
+        $finder = new Finder();
+        $finder->files()
+            ->ignoreVCS(true)
+            ->name('*.php')
+            ->notName('Compiler.php')
+            ->in(__DIR__)
+        ;
+
+        foreach ($finder as $file) {
+            $this->addFile($phar, $file);
+        }
+
+        foreach ($finder as $file) {
+            $this->addFile($phar, $file, false);
+        }
+
+        $finder = new Finder();
+        $finder->files()
+            ->ignoreVCS(true)
+            ->name('*.php')
+            ->name('LICENSE')
+            ->exclude('Tests')
+            ->exclude('tests')
+            ->in(__DIR__.'/../vendor/composer/')
+            ->in(__DIR__.'/../vendor/doctrine/annotations')
+            ->in(__DIR__.'/../vendor/doctrine/inflector')
+            ->in(__DIR__.'/../vendor/doctrine/lexer')
+            ->in(__DIR__.'/../vendor/goetas/')
+            ->in(__DIR__.'/../vendor/jms/')
+            ->in(__DIR__.'/../vendor/phpcollection/')
+            ->in(__DIR__.'/../vendor/phpoption/')
+            ->in(__DIR__.'/../vendor/symfony/')
+            ->in(__DIR__.'/../vendor/zendframework/')
+        ;
+
+        foreach ($finder as $file) {
+            $this->addFile($phar, $file);
+        }
+
+        $this->addFile($phar, new \SplFileInfo(__DIR__.'/../vendor/autoload.php'));
```

Comment:
does it make sense to include all files in `conmposer`? otherwise this feature will be fragile....